.github/workflows/ContinuousIntegration.yml

```diff
@@ -0,0 +1,34 @@
+# GitHub workflow to enable continuous integratoin
+name: Continuous Integration
+
+# This workflow is triggered on pushes and pull requests to the repository.
+on: 
+  pull_request: 
+    branches: 'master'
+
+jobs:
+  build:
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ ubuntu-18.04, macos-10.15 ]
+        cxx: [ g++-9, clang++ ]
+        build_type: [ Debug, Release ]
+
+    steps:
+    - name: which CXX
+      run: |
+        which ${{matrix.cxx}} 
+        ${{matrix.cxx}} --version
+    - uses: actions/checkout@v2
+    - name: mkdir bin
+      run: mkdir bin 
+    - name: cmake
+      run: cmake -D CMAKE_CXX_COMPILER=`which ${{matrix.cxx}}` -D CMAKE_BUILD_TYPE=${{matrix.build_type}} ..  
+      working-directory: ./bin
+    - name: make
+      run: make
+      working-directory: ./bin
+    - name: ctest
+      run: ctest -j2
```

Comment:
So just an observation: you use make with a single processor and ctest with 2. Do we know what type of machine these are being run on? It would be nice to take advantage of multiprocessing if it is available - especially for the more time intensive projects.

Reply:
Good point. I believe we have two processors at our disposal.